man/zram-generator.md

```diff
@@ -0,0 +1,57 @@
+zram-generator(8) -- Systemd unit generator for zram swap devices
+=================================================================
+
+## SYNOPSIS
+
+`/lib/systemd/system-generators/zram-generator` `TARGET_DIR` [*2RGET_DIR* *3RGET_DIR*]<br />
+`/lib/systemd/system-generators/zram-generator` --setup-device `DEVICE`<br />
+`/lib/systemd/system-generators/zram-generator` --reset-device `DEVICE`
+
+## DESCRIPTION
+
+`zram-generator` is a generator that creates zram devices and systemd units to configure, format, and enable swap on them.
+
+The generator will be invoked by systemd early at boot, the generator will then:
+
+  1. read configuration files from *{/etc,/lib}/systemd/zram-generator.conf[.d]* (see zram-generator.conf(5) for details);
+  2. if systemd-detect-virt(1) exists and returns non-*0*, exit, since continuing would mean changing the host system;
+  3. generate systemd.swap(5) units into `TARGET_DIR` for all configured devices and symlink them into *swap.target.wants/*;
+  4. if any devices are enabled, and the zram module isn't loaded (*/sys/class/zram-control* doesn't exist), invoke *modprobe(8) zram*;
+  5. if the requested devices aren't present, read */sys/class/zram-control/hot_add* until the highest device's number is read,
```

Comment:
Same for 4–5: maybe just "The generator will try to ensure that the `zram` module is loaded.".

Reply:
Updated